src/ChannelServer/World/Map.cs

```diff
@@ -109,6 +151,8 @@ public void RemoveCharacter(Character character)
 				_characters.Remove(character.Handle);
 
 			character.Map = null;
+
+			SectorManager.Remove(character);
```

Comment:
While it's unlikely to happen, *if* a tick would occur between setting Map to null and the entity actually being removed from the sector manager, a visit event could make use of Map, causing a null exception.